frontend/online-shop/src/components/productDetails/ProductDetails.tsx

```diff
@@ -0,0 +1,54 @@
+import { useStyle } from "../../styles/productDetails/ProductDetailsStyle";
+import { Grid, IconButton } from "@material-ui/core";
+import ShoppingCart from "@material-ui/icons/ShoppingCart";
+import React from "react";
+
+export const ProductDetails = (props: any) => {
+  const style = useStyle();
+
+  const { number } = props.match.params;
+  const { product } = props.location.state;
+
+  return (
+    <Grid container className={style.root}>
+      <Grid className={style.info} item xs={6}>
+        <Grid className={style.grid} item xs={12}>
```

Comment:
If you want to have the items one under the other on all the resolutions you don't have to create a `Grid` item with 12 columns for each one of them. You could set `display: flex; flex-direction: column` on their parent and they will automatically be placed one under the other. Here is a nice intro to flexboxes: [Flex boxes](https://css-tricks.com/snippets/css/a-guide-to-flexbox/)